optax/_src/transform_test.py

```diff
@@ -202,6 +202,16 @@ def test_centralize(self, inputs, outputs):
     centralized_inputs, _ = centralizer.update(inputs, None)
     chex.assert_tree_all_close(centralized_inputs, outputs)
 
+  def test_sherman_morrisson(self):
+    x = 1
+    a_inv = jnp.eye(3) * 1.0
+    u = v = jnp.ones(3)
```

Comment:
All of these test values look like special cases to me. Could you add a test with more general conditions? (If the special case is important to test please keep it around as a separate test.)